### ISSUE TITLE
Test for equality of strings

### ISSUE THREAD

**Issue by parrotrueper** [koalaman/shellcheck#2856]
#### For bugs
- Rule Id SC2193
- My shellcheck version 0.8.0

comparison of string variables should allow '='

#### Here's a snippet or screenshot that shows the problem:

```sh

if [[ "{$1,,}" = "foo" ]]; then      
#                        https://www.gnu.org/software/bash/manual/bash.html#Conditional-Constructs
#          string1 = string2
#              True if the strings are equal. When used with the [[ command, this performs pattern matching.
#
#           ‘=’ should be used with the test command for POSIX conformance.
fi

if [ "{$1,,}" = "foo" ]; then      
# same problem here
fi
```

#### Here's what shellcheck currently says:
The arguments to this comparison can never be equal.

#### Here's what I wanted or expected to see:
no porblem with '=' as per bash reference manual

**Comment by brother** [comment]
The left hand string will never be foo because no brace expansion will be performed. If we assume that $1 is set to foo you are comparing `curly brace foo comma comma curly brace` (`{foo,,}` ) with `foo` and they will never match.

```sh
brother ~$ cat /tmp/test.sh
#!/bin/bash

echo "1 is ->$1<-"

if [[ "{$1,,}" = "foo" ]]; then
	echo "Nice. 1 was set to foo."
fi

set -x

if [[ "{$1,,}" = "foo" ]]; then
	echo "Nice. 1 was set to foo."
fi
brother ~$ /tmp/test.sh foo
1 is ->foo<-
+ [[ {foo,,} = \f\o\o ]]
```

**Comment by parrotrueper** [comment]
![shellcheck](https://github.com/koalaman/shellcheck/assets/56156306/e5cd0dbc-5d1a-4243-b068-3464f9f72281)
